### PR TITLE
Re-Arrange the FORCE repo for testing

### PR DESCRIPTION
**Pull request by BotrosHanna-INL**

### CODE REVIEW

**Comment by BotrosHanna-INL** [comment]
Do not merge yet please.

**Comment by BotrosHanna-INL** [comment]
@PaulTalbot-INL 

Hi @dylanjm,
The PR is ready for review now. Now, all the Aspen/HERON integration tests are setup and are successful.
The main full test is the: /integration_tests/Aspen_HERON_FullTest.
All the others tests are tests for intermediate steps.
The tests include running python files, comparing texts, images and xml files, running HERON and RAVEN.

For now,  The tests should work if RAVEN, HERON and FORCE are inside the same folder. Some changes to the paths are needed once we have a clear idea about the structure of the FORCE installation that includes RAVEN, HERON, HYBRID, etc.

For the tests to work, the raven libraries need to be activated: "conda activate raven_libraries"
Thank you!

Pasted Graphic

File: ravenframework/MessageHandler.py
Comment (PaulTalbot-INL) [inline ravenframework/MessageHandler.py:1]: It looks like a lot of RAVEN got added to this PR; is that intentional? I recognize we still need to put the intelligence for pointing to RAVEN in, but we probably shouldn't be adding RAVEN files directly into FORCE.

File: rook/.gitignore
Comment (PaulTalbot-INL) [inline rook/.gitignore:1]: Similar for rook, we should be looping in an existing rook, not directly adding it to the repo, I think.

File: rook/__init__.py
Comment (PaulTalbot-INL) [inline rook/__init__.py:1]: I think this file should still be in vertical_integration/src/, right?

File: run_tests [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# Note that this script is taken from raven "raven/run_tests". Some changes were made to make
```
Comment (PaulTalbot-INL) [inline run_tests:1]: this note should probably be finished or removed

File: run_tests [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+SCRIPT_NAME=`readlink $0`
+if test -x "$SCRIPT_NAME";
+then
+    SCRIPT_DIRNAME=`dirname $SCRIPT_NAME`
+else
+    SCRIPT_DIRNAME=`dirname $0`
+fi
+SCRIPT_DIR=`(cd $SCRIPT_DIRNAME; pwd)`
+
+TEST_SET=0 # 0 for normal tests, 1 for plugin tests, 2 for both
```
Comment (PaulTalbot-INL) [inline run_tests:14]: needed for FORCE tests?

**Review by PaulTalbot-INL — CHANGES_REQUESTED** [review]
It's great to hear these tests are set up and can run! However, it looks like a lot of RAVEN and ROOK got copied into FORCE and added to this PR. Those additions should be removed, unless I'm mistaken.

**Comment by PaulTalbot-INL** [comment]
I didn't perform a full review, as there's quite a bit of extra added to this PR at this point that doesn't need reviewing.

**Review by joshua-cogliati-inl — COMMENTED** [review]
Some comments from looking at this.

File: ravenframework/MessageHandler.py
Comment (BotrosHanna-INL) [reply ravenframework/MessageHandler.py:1]: I have most of the raven files. What we have now from raven is the rook file only: "FORCE/developer_tools/rook.ini"

File: run_tests (hunk 1 above)
Comment (BotrosHanna-INL) [reply run_tests:1]: Fixed it.

File: run_tests (hunk 2 above)
Comment (BotrosHanna-INL) [reply run_tests]: No, this is not needed. I cleaned up the run_tests file and the run_tests file is much smaller now.

File: run_tests [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+if [[ $TEST_SET == 1 ]] || [[ $TEST_SET == 2 ]]; then
+  DO_PLUGINS=0
+else
+  DO_PLUGINS=1
+fi
+
+for A in "$@"; do
+    case $A in
+        --library_report | --library-report)
+            $PYTHON_COMMAND $SCRIPT_DIR/scripts/library_report
```
[the remark this answers is not in the document]
Comment (BotrosHanna-INL) [reply run_tests]: I cleaned up the run_tests file.

**Comment by BotrosHanna-INL** [comment]
@PaulTalbot-INL 
Hi Paul, let me know if more changes are needed to this PR.

**Review by dylanjm — APPROVED** [review]
Changes are good and new folder structure will help keep everything organized.

**Pull request merged**